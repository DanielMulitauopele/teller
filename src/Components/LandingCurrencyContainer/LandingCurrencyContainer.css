.currency-container {
  margin-top: 10rem;
  width: 95%;
  overflow: scroll;
  position: absolute;
  right: 0;
  top: 1rem;
  height: 72vh;
  -webkit-overflow-scrolling: touch;
  z-index: 2;
}

.sort-box {
  text-align: left;
}

.sort {
  text-align: right;
  position: sticky;
  top: 0rem;
  background-color: var(--TellerBG);
  right: 0;
  z-index: 4;
  width: 100%;
}

.sort a {
  color: white;
  text-transform: uppercase;
  font-size: 0.8rem;
  letter-spacing: 1px;
  margin-right: 1rem;
  text-decoration: none;
  display: inline-block;
  border-bottom: 3px solid transparent;
  transition: 0.3s;
  margin: 1rem 2rem;
}

.sort-box a:hover {
  border-bottom: 3px solid #fc599f;
}
