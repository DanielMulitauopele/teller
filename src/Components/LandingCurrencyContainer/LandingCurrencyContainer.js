import React, { Component } from "react";
import "./LandingCurrencyContainer.css";
import LandingCurrency from "../LandingCurrency/LandingCurrency";

class LandingCurrencyContainer extends Component {
  constructor({ props, abbrevCurrencies, setFilter, addToFavorites }) {
    super(props);
    this.state = {
      active: false
    };
  }

  handleClick = e => {
    const { name } = e.target;
    this.props.setFilter(name);
  };

  render() {
    const { abbrevCurrencies, addToFavorites } = this.props;
    const landingCurrencies = abbrevCurrencies.map((currency, i) => {
      return (
        <LandingCurrency
          key={i}
          index={i}
          currency={currency}
          addToFavorites={addToFavorites}
        />
      );
    });

    return (
      <div className="sort-box">
        <div className="currency-container">
          <div className="sort">
            <button
              className="rank-link"
              // href="javascript:void(0)"
              name="Rank"
              onClick={this.handleClick}
            >
              Rank
            </button>
            <button
              className="price-link"
              // href="javascript:void(0)"
              name="Price"
              onClick={this.handleClick}
            >
              Price
            </button>
            <button
              className="percent-change-link"
              // href="javascript:void(0)"
              name="%Change"
              onClick={this.handleClick}
            >
              %Change
            </button>
          </div>
          {landingCurrencies}
        </div>
      </div>
    );
  }
}

export default LandingCurrencyContainer;
