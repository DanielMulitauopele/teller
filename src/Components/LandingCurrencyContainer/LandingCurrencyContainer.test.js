import React from 'react'
import { shallow } from 'enzyme'
import LandingCurrencyContainer from './LandingCurrencyContainer'
import { mockAbbrevCurrencies } from '../../Utils/MockData/mockAbbrevCurrencies'

describe('LandingCurrencyContainer Component', () => {
  let wrapper
  const mockSetFilter = jest.fn()
  const mockAddToFavorites = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<LandingCurrencyContainer
                        abbrevCurrencies={mockAbbrevCurrencies}
                        setFilter={mockSetFilter}
                        addToFavorites={mockAddToFavorites}
                      />)
  })

  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should start with the correct default state', () => {
    expect(wrapper.state().active).toEqual(false)
  })

  describe('handleClick function', () => {
    it('should call setFilter with the correct params', () => {
      const rankLink = wrapper.find('.rank-link')
      rankLink.simulate('click', {
        target: {name: 'Rank'}
      })
      expect(mockSetFilter).toHaveBeenCalledWith('Rank')
    })
  })
})