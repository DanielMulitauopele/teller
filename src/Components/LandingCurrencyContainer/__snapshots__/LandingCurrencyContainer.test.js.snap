// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCurrencyContainer Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingCurrencyContainer
    abbrevCurrencies={
      Array [
        Object {
          "name": "Bitcoin",
          "percent_change": 0.83,
          "price": 3586.16,
          "rank": "1",
          "symbol": "BTC",
        },
        Object {
          "name": "XRP",
          "percent_change": 3.94,
          "price": 0.33,
          "rank": "2",
          "symbol": "XRP",
        },
        Object {
          "name": "Ethereum",
          "percent_change": 0.2,
          "price": 95.07,
          "rank": "3",
          "symbol": "ETH",
        },
        Object {
          "name": "EOS",
          "percent_change": 6.87,
          "price": 2.54,
          "rank": "4",
          "symbol": "EOS",
        },
        Object {
          "name": "Stellar",
          "percent_change": -0.58,
          "price": 0.11,
          "rank": "5",
          "symbol": "XLM",
        },
        Object {
          "name": "Tether",
          "percent_change": -0.37,
          "price": 1.02,
          "rank": "6",
          "symbol": "USDT",
        },
        Object {
          "name": "Bitcoin Cash",
          "percent_change": 12.32,
          "price": 100.65,
          "rank": "7",
          "symbol": "BCH",
        },
        Object {
          "name": "Litecoin",
          "percent_change": -1.97,
          "price": 28.99,
          "rank": "8",
          "symbol": "LTC",
        },
        Object {
          "name": "Bitcoin SV",
          "percent_change": -1.04,
          "price": 81.84,
          "rank": "9",
          "symbol": "BSV",
        },
        Object {
          "name": "TRON",
          "percent_change": 0.17,
          "price": 0.01,
          "rank": "10",
          "symbol": "TRX",
        },
      ]
    }
    addToFavorites={[MockFunction]}
    setFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="currency-container"
      >
        <div
          className="sort"
        >
          <a
            className="rank-link"
            href="#"
            name="Rank"
            onClick={[Function]}
          >
            Rank
          </a>
          <a
            className="price-link"
            href="#"
            name="Price"
            onClick={[Function]}
          >
            Price
          </a>
          <a
            className="percent-change-link"
            href="#"
            name="%Change"
            onClick={[Function]}
          >
            %Change
          </a>
        </div>
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Bitcoin",
              "percent_change": 0.83,
              "price": 3586.16,
              "rank": "1",
              "symbol": "BTC",
            }
          }
          index={0}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "XRP",
              "percent_change": 3.94,
              "price": 0.33,
              "rank": "2",
              "symbol": "XRP",
            }
          }
          index={1}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Ethereum",
              "percent_change": 0.2,
              "price": 95.07,
              "rank": "3",
              "symbol": "ETH",
            }
          }
          index={2}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "EOS",
              "percent_change": 6.87,
              "price": 2.54,
              "rank": "4",
              "symbol": "EOS",
            }
          }
          index={3}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Stellar",
              "percent_change": -0.58,
              "price": 0.11,
              "rank": "5",
              "symbol": "XLM",
            }
          }
          index={4}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Tether",
              "percent_change": -0.37,
              "price": 1.02,
              "rank": "6",
              "symbol": "USDT",
            }
          }
          index={5}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Bitcoin Cash",
              "percent_change": 12.32,
              "price": 100.65,
              "rank": "7",
              "symbol": "BCH",
            }
          }
          index={6}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Litecoin",
              "percent_change": -1.97,
              "price": 28.99,
              "rank": "8",
              "symbol": "LTC",
            }
          }
          index={7}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "Bitcoin SV",
              "percent_change": -1.04,
              "price": 81.84,
              "rank": "9",
              "symbol": "BSV",
            }
          }
          index={8}
        />
        <LandingCurrency
          addToFavorites={[MockFunction]}
          currency={
            Object {
              "name": "TRON",
              "percent_change": 0.17,
              "price": 0.01,
              "rank": "10",
              "symbol": "TRX",
            }
          }
          index={9}
        />
      </div>,
      "className": "sort-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sort"
          >
            <a
              className="rank-link"
              href="#"
              name="Rank"
              onClick={[Function]}
            >
              Rank
            </a>
            <a
              className="price-link"
              href="#"
              name="Price"
              onClick={[Function]}
            >
              Price
            </a>
            <a
              className="percent-change-link"
              href="#"
              name="%Change"
              onClick={[Function]}
            >
              %Change
            </a>
          </div>,
          Array [
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Bitcoin",
                  "percent_change": 0.83,
                  "price": 3586.16,
                  "rank": "1",
                  "symbol": "BTC",
                }
              }
              index={0}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "XRP",
                  "percent_change": 3.94,
                  "price": 0.33,
                  "rank": "2",
                  "symbol": "XRP",
                }
              }
              index={1}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Ethereum",
                  "percent_change": 0.2,
                  "price": 95.07,
                  "rank": "3",
                  "symbol": "ETH",
                }
              }
              index={2}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "EOS",
                  "percent_change": 6.87,
                  "price": 2.54,
                  "rank": "4",
                  "symbol": "EOS",
                }
              }
              index={3}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Stellar",
                  "percent_change": -0.58,
                  "price": 0.11,
                  "rank": "5",
                  "symbol": "XLM",
                }
              }
              index={4}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Tether",
                  "percent_change": -0.37,
                  "price": 1.02,
                  "rank": "6",
                  "symbol": "USDT",
                }
              }
              index={5}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Bitcoin Cash",
                  "percent_change": 12.32,
                  "price": 100.65,
                  "rank": "7",
                  "symbol": "BCH",
                }
              }
              index={6}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Litecoin",
                  "percent_change": -1.97,
                  "price": 28.99,
                  "rank": "8",
                  "symbol": "LTC",
                }
              }
              index={7}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "Bitcoin SV",
                  "percent_change": -1.04,
                  "price": 81.84,
                  "rank": "9",
                  "symbol": "BSV",
                }
              }
              index={8}
            />,
            <LandingCurrency
              addToFavorites={[MockFunction]}
              currency={
                Object {
                  "name": "TRON",
                  "percent_change": 0.17,
                  "price": 0.01,
                  "rank": "10",
                  "symbol": "TRX",
                }
              }
              index={9}
            />,
          ],
        ],
        "className": "currency-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="rank-link"
                href="#"
                name="Rank"
                onClick={[Function]}
              >
                Rank
              </a>,
              <a
                className="price-link"
                href="#"
                name="Price"
                onClick={[Function]}
              >
                Price
              </a>,
              <a
                className="percent-change-link"
                href="#"
                name="%Change"
                onClick={[Function]}
              >
                %Change
              </a>,
            ],
            "className": "sort",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rank",
                "className": "rank-link",
                "href": "#",
                "name": "Rank",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Rank",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Price",
                "className": "price-link",
                "href": "#",
                "name": "Price",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Price",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "%Change",
                "className": "percent-change-link",
                "href": "#",
                "name": "%Change",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "%Change",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Bitcoin",
              "percent_change": 0.83,
              "price": 3586.16,
              "rank": "1",
              "symbol": "BTC",
            },
            "index": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "XRP",
              "percent_change": 3.94,
              "price": 0.33,
              "rank": "2",
              "symbol": "XRP",
            },
            "index": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Ethereum",
              "percent_change": 0.2,
              "price": 95.07,
              "rank": "3",
              "symbol": "ETH",
            },
            "index": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "EOS",
              "percent_change": 6.87,
              "price": 2.54,
              "rank": "4",
              "symbol": "EOS",
            },
            "index": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Stellar",
              "percent_change": -0.58,
              "price": 0.11,
              "rank": "5",
              "symbol": "XLM",
            },
            "index": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Tether",
              "percent_change": -0.37,
              "price": 1.02,
              "rank": "6",
              "symbol": "USDT",
            },
            "index": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Bitcoin Cash",
              "percent_change": 12.32,
              "price": 100.65,
              "rank": "7",
              "symbol": "BCH",
            },
            "index": 6,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Litecoin",
              "percent_change": -1.97,
              "price": 28.99,
              "rank": "8",
              "symbol": "LTC",
            },
            "index": 7,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "Bitcoin SV",
              "percent_change": -1.04,
              "price": 81.84,
              "rank": "9",
              "symbol": "BSV",
            },
            "index": 8,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "class",
          "props": Object {
            "addToFavorites": [MockFunction],
            "currency": Object {
              "name": "TRON",
              "percent_change": 0.17,
              "price": 0.01,
              "rank": "10",
              "symbol": "TRX",
            },
            "index": 9,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="currency-container"
        >
          <div
            className="sort"
          >
            <a
              className="rank-link"
              href="#"
              name="Rank"
              onClick={[Function]}
            >
              Rank
            </a>
            <a
              className="price-link"
              href="#"
              name="Price"
              onClick={[Function]}
            >
              Price
            </a>
            <a
              className="percent-change-link"
              href="#"
              name="%Change"
              onClick={[Function]}
            >
              %Change
            </a>
          </div>
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              }
            }
            index={0}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              }
            }
            index={1}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              }
            }
            index={2}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              }
            }
            index={3}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              }
            }
            index={4}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              }
            }
            index={5}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              }
            }
            index={6}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              }
            }
            index={7}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              }
            }
            index={8}
          />
          <LandingCurrency
            addToFavorites={[MockFunction]}
            currency={
              Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              }
            }
            index={9}
          />
        </div>,
        "className": "sort-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sort"
            >
              <a
                className="rank-link"
                href="#"
                name="Rank"
                onClick={[Function]}
              >
                Rank
              </a>
              <a
                className="price-link"
                href="#"
                name="Price"
                onClick={[Function]}
              >
                Price
              </a>
              <a
                className="percent-change-link"
                href="#"
                name="%Change"
                onClick={[Function]}
              >
                %Change
              </a>
            </div>,
            Array [
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Bitcoin",
                    "percent_change": 0.83,
                    "price": 3586.16,
                    "rank": "1",
                    "symbol": "BTC",
                  }
                }
                index={0}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "XRP",
                    "percent_change": 3.94,
                    "price": 0.33,
                    "rank": "2",
                    "symbol": "XRP",
                  }
                }
                index={1}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Ethereum",
                    "percent_change": 0.2,
                    "price": 95.07,
                    "rank": "3",
                    "symbol": "ETH",
                  }
                }
                index={2}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "EOS",
                    "percent_change": 6.87,
                    "price": 2.54,
                    "rank": "4",
                    "symbol": "EOS",
                  }
                }
                index={3}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Stellar",
                    "percent_change": -0.58,
                    "price": 0.11,
                    "rank": "5",
                    "symbol": "XLM",
                  }
                }
                index={4}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Tether",
                    "percent_change": -0.37,
                    "price": 1.02,
                    "rank": "6",
                    "symbol": "USDT",
                  }
                }
                index={5}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Bitcoin Cash",
                    "percent_change": 12.32,
                    "price": 100.65,
                    "rank": "7",
                    "symbol": "BCH",
                  }
                }
                index={6}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Litecoin",
                    "percent_change": -1.97,
                    "price": 28.99,
                    "rank": "8",
                    "symbol": "LTC",
                  }
                }
                index={7}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "Bitcoin SV",
                    "percent_change": -1.04,
                    "price": 81.84,
                    "rank": "9",
                    "symbol": "BSV",
                  }
                }
                index={8}
              />,
              <LandingCurrency
                addToFavorites={[MockFunction]}
                currency={
                  Object {
                    "name": "TRON",
                    "percent_change": 0.17,
                    "price": 0.01,
                    "rank": "10",
                    "symbol": "TRX",
                  }
                }
                index={9}
              />,
            ],
          ],
          "className": "currency-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="rank-link"
                  href="#"
                  name="Rank"
                  onClick={[Function]}
                >
                  Rank
                </a>,
                <a
                  className="price-link"
                  href="#"
                  name="Price"
                  onClick={[Function]}
                >
                  Price
                </a>,
                <a
                  className="percent-change-link"
                  href="#"
                  name="%Change"
                  onClick={[Function]}
                >
                  %Change
                </a>,
              ],
              "className": "sort",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Rank",
                  "className": "rank-link",
                  "href": "#",
                  "name": "Rank",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Rank",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Price",
                  "className": "price-link",
                  "href": "#",
                  "name": "Price",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Price",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "%Change",
                  "className": "percent-change-link",
                  "href": "#",
                  "name": "%Change",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "%Change",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              },
              "index": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              },
              "index": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              },
              "index": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              },
              "index": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              },
              "index": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              },
              "index": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              },
              "index": 6,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              },
              "index": 7,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              },
              "index": 8,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "class",
            "props": Object {
              "addToFavorites": [MockFunction],
              "currency": Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              },
              "index": 9,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
